bscript/oppushdata.go

```diff
@@ -90,37 +90,51 @@ func DecodeParts(b []byte) ([][]byte, error) {
 			if len(b) < 2 {
 				return r, errors.New("not enough data")
 			}
-			l := uint64(b[1])
-			if len(b) < int(2+l) {
+
+			l := int(b[1])
+			b = b[2:]
+ 
+			if len(b) < l {
 				return r, errors.New("not enough data")
 			}
-			part := b[2 : 2+l]
+
+			part := b[:l]
 			r = append(r, part)
-			b = b[2+l:]
+			b = b[l:]
 
 		case OpPUSHDATA2:
 			if len(b) < 3 {
 				return r, errors.New("not enough data")
 			}
-			l := binary.LittleEndian.Uint16(b[1:])
-			if len(b) < int(3+l) {
+
+			l := int(binary.LittleEndian.Uint16(b[1:]))
```

Comment:
Would probably be better to use  binary.LittleEndian.Uint64(b[1:])  here and it should resolve this issue without needing to convert to int etc